frontend/src/pages/Users.tsx

```diff
@@ -0,0 +1,63 @@
+import { useNavigate } from 'react-router-dom'
+import FontAwesomeIconWrapper from 'lib/FontAwesomeIconWrapper'
+import { useSearchPage } from 'lib/hooks/useSearchPage'
+import { user } from 'lib/types'
+import SearchPageLayout from 'components/SearchPageLayout'
+import UserCard from 'components/Usercard'
+
+const UsersPage = () => {
+  const {
+    items: users,
+    isLoaded,
+    currentPage,
+    totalPages,
+    searchQuery,
+    handleSearch,
+    handlePageChange,
+  } = useSearchPage<user>({
+    indexName: 'users',
+    pageTitle: 'OWASP Users',
+  })
+
+  const navigate = useNavigate()
+
+  const handleButtonClick = (user: user) => {
+    navigate(`/users/${user.idx_key}`)
+  }
+
+  const renderUserCard = (user: user) => {
+    const SubmitButton = {
+      label: 'View Details',
+      icon: <FontAwesomeIconWrapper icon="fa-solid fa-right-to-bracket" />,
+      onclick: () => handleButtonClick(user),
+    }
+
+    return (
+      <UserCard
+        avatar={user.idx_avatar_url}
+        name={user.idx_name}
+        company={user.idx_company}
+        button={SubmitButton}
+      />
+    )
+  }
+
+  return (
+    <SearchPageLayout
+      isLoaded={isLoaded}
+      totalPages={totalPages}
+      currentPage={currentPage}
+      searchQuery={searchQuery}
+      onSearch={handleSearch}
+      onPageChange={handlePageChange}
+      empty="No Users found"
+      searchPlaceholder="Search for OWASP users..."
+    >
+      <div className="grid grid-cols-1 gap-6 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
```

Comment:
It would be nice to make these responsive right from the start. Right now on iPadPro or iPad Air view there's no padding between cards. Also, there's no padding on the right side of the page on that view. 🤔
Can we make sure all major sizes are fully covered with these classes?
Some examples of what I'm seeing right now:
![Screenshot 2025-01-05 at 5 35 51 PM](https://github.com/user-attachments/assets/9e396d89-3c49-4f16-907e-0f4dba7836ac)
![Screenshot 2025-01-05 at 5 35 57 PM](https://github.com/user-attachments/assets/2ad20389-6914-4351-95e0-3a5d82a4ecdd)